### PR TITLE
fix unnecessary re-renders

### PR DESCRIPTION
**Pull request by Nerevar123**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Тow, even if handlers for node events are not passed, events are processed and unnecessary re-renders occur

## What is the new behavior?

Added check if event handler is passed

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/symbols/Node/Node.tsx [hunk 1]
```diff
@@ -139,10 +147,10 @@ export const Node: FC<Partial<NodeProps>> = ({
   onDrag = () => undefined,
   onDragStart = () => undefined,
   onDragEnd = () => undefined,
-  onClick = () => undefined,
-  onKeyDown = () => undefined,
-  onEnter = () => undefined,
-  onLeave = () => undefined
+  onClick = null,
```
Comment (amcdnl) [inline src/symbols/Node/Node.tsx:150]: We could probably kill these now and use `onClick.?()` operators

File: src/symbols/Node/Node.tsx [hunk 2]
```diff
@@ -266,37 +274,49 @@ export const Node: FC<Partial<NodeProps>> = ({
...
-        }}
-        onClick={event => {
-          event.preventDefault();
-          event.stopPropagation();
-          if (!isDisabled && canSelect) {
-            onClick(event, properties);
-          }
-        }}
+        onKeyDown={
+          onKeyDown &&
```
Comment (amcdnl) [inline src/symbols/Node/Node.tsx:278]: Your PR says fix unneeded renders - if this check is the only real material change - I'm curious why this helped ( just personally for me ).

File: src/symbols/Node/Node.tsx (hunk 1 above)
Comment (Nerevar123) [reply src/symbols/Node/Node.tsx:150]: if I kill these lines - TS starts to scream

File: src/symbols/Node/Node.tsx (hunk 2 above)
Comment (Nerevar123) [reply src/symbols/Node/Node.tsx:278]: I use node hover styles and if there are more then 50 nodes - get getting significant delays with this effect.
Started researching and found out that every onMouseEnter and omMouseLeave elements are re-rendered.

You can reproduce it with any current story on documentation, just start React profiler and move the cursor over the nodes, you will see re-renders

**Comment by Nerevar123** [comment]
@amcdnl any other changes need to be made?
Waiting for continue the development with this fix 😊

**Comment by amcdnl** [comment]
@Nerevar123 - All good - merged and releasing now

**Comment by amcdnl** [comment]
We probably want to do this for the other components too - can u look to see if that would help them too?

**Comment by Nerevar123** [comment]
Thank you!

Sure, I'll take a look

**Pull request merged**